docs/en/sql_reference/aggregate_functions/reference.md

```diff
@@ -527,6 +527,47 @@ Calculates the average.
 Only works for numbers.
 The result is always Float64.
 
+
+## avgWeighted {#avgweighted}
+
+Calculates the [weighted arithmetic mean](https://en.wikipedia.org/wiki/Weighted_arithmetic_mean).
+
+**Syntax** 
+
+```sql
+avgWeighted(x, weight)
+```
+
+**Parameters** 
+
+- `x` — Values. [Integer](../data_types/int_uint.md) or [floating-point](../data_types/float.md).
+- `weight` — Weights of the values. [Integer](../data_types/int_uint.md) or [floating-point](../data_types/float.md).
+
+Type of `x` and `weight` must be the same.
+
+**Returned value**
+
+- Weighted mean. Type: [Float](../data_types/float.md).
+- `NaN`. If all the weights are equal to 0.
```

Comment:
NaN is also a floating point value, it'd look better as a comment in previous bullet or as a separate !!! note